timeseries/src/autogluon/timeseries/evaluator.py

```diff
@@ -215,10 +238,15 @@ def check_get_evaluation_metric(
 
     def save_past_metrics(self, data_past: TimeSeriesDataFrame):
         seasonal_period = get_seasonality(data_past.freq) if self.seasonal_period is None else self.seasonal_period
-        self._past_naive_error = in_sample_seasonal_naive_error(
+        self._past_abs_seasonal_error = in_sample_abs_seasonal_error(
             y_past=data_past[self.target_column], seasonal_period=seasonal_period
         )
```

Comment:
For the first point, I was suggesting the following change
```suggestion
        if self.eval_metric == "MASE":
            self._past_abs_seasonal_error = in_sample_abs_seasonal_error(
                y_past=data_past[self.target_column], seasonal_period=seasonal_period
            )
```
This would ensure that we only compute the `_past_abs_seasonal_error` for the metrics that need it (i.e. MASE), and don't spend time on this computation for other metrics.

Reply:
Done!